### PR TITLE
Fix staticcheck linting problems

### PR DESCRIPTION
**Pull request by albertony**
#### What is the purpose of this change?

Running the staticcheck lint tool (v0.3.2) and fixing reported "problems". With a clean master it reports more than 300 problems. This PR removes more than 100 of these. After this PR and also the separate #6237 which fixes "Error strings should not be capitalized" reported issues, the total number of reported problems is reduced to something close to 60.

Example of frequently reported problems are:
- Avoid unnecessary use of fmt.Sprintf (S1039)
- Use unconditional TrimPrefix instead of conditional with HasPrefix
- Use strings.EqualFold instead of comparing results of ToLower
   - Note that the result is not 100% guaranteed identical due to Unicode case-folding rules, and therefore there have been some debate on this lint rule. Not very likely to hit any real cases where there is difference? And perhaps EqualFold is closer to what is expected anyway? Primary reasons to use EqualFold is that it supposedly outperforms the ToLower comparison by orders of magnitude, and also is more readable (in my opinion).
- Merge variable declaration and assignment (S1021)
- Replace x.Sub(time.Now()) with time.Until(x) (S1024)

Remaining lint problems:
- Error strings should not be capitalized (ST1005)
    - Done in separate PR: #6237
- Some Linux mount specifics fixed in separate PRs #6238 and #6239
- The rest is mostly deprecations and unused functions.
- Some "assigning the result of this type assertion to a variable (switch e := e.(type)) could eliminate type assertions in switch cases (S1034)".
- One issue in lib\http\auth\basic.go can be worth looking into: "this value of r is never used (SA4006)" and "WithContext is a pure function but its return value is ignored (SA4017)" on line `r = r.WithContext(context.WithValue(r.Context(), ContextAuthKey, value))`
- Another issue in vfs\vfs.go: "ineffective break statement. Did you mean to break out of the outer loop? (SA4011)", on the break statement in `case <-tick.C:`

#### Checklist

- [X] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [ ] I have added tests for all changes in this PR if appropriate.
- [ ] I have added documentation for the changes if appropriate.
- [X] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [X] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: backend/netstorage/netstorage.go [hunk 1]
```diff
@@ -903,22 +903,20 @@ func (f *Fs) netStorageStatRequest(ctx context.Context, URL string, directory bo
...
-				fs.Infof(nil, "Converting a symlink to the rclonelink on the stat request %s", files[i].Name)
-			}
-			entrywanted := (directory && files[i].Type == "dir") ||
-				(!directory && files[i].Type != "dir")
-			if entrywanted {
-				filestamp := files[0]
-				files[0] = files[i]
-				files[i] = filestamp
-			}
+	// Multiple objects can be returned with the "slash=both" option,
```
[the remark this answers is not in the document]
Comment (albertony) [reply backend/netstorage/netstorage.go:906]: Thats exactly the point, according to this: https://staticcheck.io/docs/checks#S1009

File: backend/opendrive/opendrive.go [hunk 2]
```diff
@@ -611,13 +608,13 @@ func (f *Fs) Put(ctx context.Context, in io.Reader, src fs.ObjectInfo, options .
 		return nil, err
 	}
 
-	if "" == o.id {
+	if o.id == "" {
```
[the remark this answers is not in the document]
Comment (albertony) [reply backend/opendrive/opendrive.go:611]: Right... and in PowerShell null-checks are preferred with null on the left side due to some type inference rules. Anyway, staticcheck claims to represent "Go" when they say this way is more readable, and I tend to  agree... :)

> In Go, which doesn’t allow for this kind of bug, we prefer the more idiomatic if x == 42.

https://staticcheck.io/docs/checks#ST1017

File: fs/config_test.go [hunk 3]
```diff
@@ -11,7 +11,7 @@ func TestGetConfig(t *testing.T) {
 	ctx := context.Background()
 
 	// Check nil
-	config := GetConfig(nil)
+	config := GetConfig(context.TODO())
```
[the remark this answers is not in the document]
Comment (albertony) [reply fs/config_test.go:14]: Ah, right 😴

File: backend/onedrive/onedrive.go [hunk 4]
```diff
@@ -2359,6 +2358,9 @@ func (f *Fs) ChangeNotify(ctx context.Context, notifyFunc func(string, fs.EntryT
 
 func (f *Fs) changeNotifyStartPageToken(ctx context.Context) (nextDeltaToken string, err error) {
 	delta, err := f.changeNotifyNextChange(ctx, "latest")
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (albertony) [reply backend/onedrive/onedrive.go:2361]: This seems like an ancient issue in errcheck; no.7, from 2013, still open.

**Comment by ncw** [comment]
> > Are you going to propose a staticcheck stage for the build at some point?
> 
> I'm dreaming of it... :) But haven't though too much about it, yet. First task is to see if I can get number of reported "problems" down to a reasonable level. And then would probably need quite a few staticcheck-specific ignore comments or similar to silence the rest (e.g. false positives), but shouldn't be a huge problem, as long as we agree that staticcheck is worth it (annoying with code that is run through x number of different linters requiring x number of specially formatted ignore comments on a single "problem").

Static check has found some real problems, and the deprecated checks are useful.

We could put it in as an advisory to start with, so run in in the `lint` build but don't fail the build if it fails. Though we'd need to fix all the warnings first, so if we get that far maybe that is pointless.

I note `golangci-lint`: https://golangci-lint.run/usage/quick-start/ does have

> staticcheck (megacheck): It's a set of rules from staticcheck. It's not the same thing as the staticcheck binary. The author of staticcheck doesn't support or approve the use of staticcheck as a library inside golangci-lint. [fast: false, auto-fix: false]

So maybe we can get a lot of the benefit just enabling it in `.golangci-lint.yml`?

**Pull request merged**